### PR TITLE
Compute

### PR DESCRIPTION
**Pull request by yujunliang**

### CODE REVIEW

File: chaosgcp/compute/actions.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 Google LLC.
```
Comment (Lawouach) [inline chaosgcp/compute/actions.py:1]: Please consider removing. Unless this is mandatory on your side.

File: chaosgcp/compute/actions.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# SPDX-License-Identifier: Apache-2.0
+from typing import Any, Dict
+
+from chaoslib.exceptions import ActivityFailed
+from chaoslib.types import Configuration, Secrets
+from google.cloud import compute_v1
+from google.cloud.compute_v1.types import Tags
+from logzero import logger
```
Comment (Lawouach) [inline chaosgcp/compute/actions.py]: This has been dropped in favour of directly using the `logging` module

```python
import logging
from typing import Any, Dict

```

Then:


```python
__all__ = ["set_instance_tags"]
logger = logging.getLogger("chaostoolkit")
```

File: chaosgcp/compute/actions.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+from chaosgcp import get_context, load_credentials, wait_on_extended_operation
+
+__all__ =["set_instance_tags"]
```
Comment (Lawouach) [inline chaosgcp/compute/actions.py]: add missing space after `=`

File: chaosgcp/compute/actions.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+def set_instance_tags(project_id:str, zone:str,instance_name: str,tags_list: list, configuration: Configuration = None,
+                              secrets: Secrets = None) -> Dict[str, Any]:
```
Comment (Lawouach) [inline chaosgcp/compute/actions.py]: lint the signature

File: chaosgcp/compute/actions.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
```
Comment (Lawouach) [inline chaosgcp/compute/actions.py]: add missing docstring

File: chaosgcp/compute/actions.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        zone=zone,
+        tags_resource=Tags(fingerprint=fgp,items=tags_list)
+                
+    )
+
+    # Make the request
+    operation = client.set_tags(request=request)
+
+    # Handle the response
+    wait_on_extended_operation(operation)
```
Comment (Lawouach) [inline chaosgcp/compute/actions.py:57]: missing a return statement

File: chaosgcp/compute/actions.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+__all__ =["set_instance_tags"]
+
+def set_instance_tags(project_id:str, zone:str,instance_name: str,tags_list: list, configuration: Configuration = None,
+                              secrets: Secrets = None) -> Dict[str, Any]:
+    
+    credentials = load_credentials(secrets)
+    ctx = get_context(configuration, secrets)
+    # Create a client
+    client = compute_v1.InstancesClient()
```
Comment (Lawouach) [inline chaosgcp/compute/actions.py]: Pass the credentials to the function

```python
client = compute_v1.InstancesClient(credentials=credentials)
```

File: tests/test_compute.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    
+
+    instance_req.return_value=compute_v1.GetInstanceRequest(instance=instance_name, project=project_id,zone=zone_name)
+    instance_req.assert_called_with(instance=instance_name, project=project_id,zone=zone_name)
+    client.get.return_value=fixtures
+    settag_req.return_value=compute_v1.SetTagsInstanceRequest(instance=instance_name,
+        project=project_id,
+        zone=zone_name,
+        tags_resource=tags(fingerprint=fixtures.compute.response_get["tags"]["fingerprint"],items=tags_list))
+    settag_req.assert_called_with(instance=instance_name, project=project_id,zone=zone_name,tags_resource=tags(fingerprint=fixtures.compute.response_get["tags"]["fingerprint"],items=tags_list))
```
Comment (Lawouach) [inline tests/test_compute.py]: lint to a column size of 80 max

**Review by Lawouach — CHANGES_REQUESTED** [review]
Please adjust as per the comments.

Also:

* add a changelog entry
* add the signature to the commits as per the DCO

**Review by Lawouach — CHANGES_REQUESTED** [review]
Hi @yujunliang,

Thank you for the update.

It contains a few more linting errors https://github.com/chaostoolkit-incubator/chaostoolkit-google-cloud-platform/actions/runs/8803403285/job/24161698162

You can ignore the errors regarding building on MacOSX. I'll fix that once we have merged.

File: chaosgcp/compute/actions.py (hunk 5 above)
Comment (yujunliang) [reply chaosgcp/compute/actions.py]: Done

File: chaosgcp/compute/actions.py (hunk 6 above)
Comment (Lawouach) [reply chaosgcp/compute/actions.py:57]: You are still missing that return statement :)

File: tests/test_compute.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 Google LLC.
+# SPDX-License-Identifier: Apache-2.0
+Signed-off-by: Yujun Liang <yujunl@google.com>
```
Comment (Lawouach) [inline tests/test_compute.py]: I think that line is likely nor where it should. please remove

**Review by Lawouach — CHANGES_REQUESTED** [review]
a couple of changes

File: tests/test_compute.py (hunk 9 above)
Comment (yujunliang) [reply tests/test_compute.py]: Removed.

**Pull request merged**